packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -247,6 +252,17 @@ export const Table = memo((props: Props) => {
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  // Set page based on useLocation() hook for current table uid
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    const tablePage = params.get(`table${uid}Page`);
+    if (tablePage) {
+      const pageNumber = Number(tablePage) - 1;
+      gotoPage(pageNumber);
+    }
+  }, [location, uid, gotoPage]);
+
+  // TODO update how reset occurs so it does not bypass persistance
```

Comment:
`if (data.length / pageSize < state.pageIndex) {
      gotoPage(0);
    }`

This calculation is having some issues with larger page numbers.

Reply:
@codeincarnate for your reference